go/test/endtoend/vreplication/vreplication_vtctldclient_cli_test.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vreplication
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"vitess.io/vitess/go/test/endtoend/cluster"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"vitess.io/vitess/go/vt/proto/binlogdata"
+	"vitess.io/vitess/go/vt/proto/vschema"
+	"vitess.io/vitess/go/vt/proto/vtctldata"
+)
+
+// TestVtctldclientCLI tests the vreplication vtctldclient CLI commands, primarily to check that non-standard flags
+// are being handled correctly. The other end-to-end tests are expected to test the various common workflows.
+func TestVtctldclientCLI(t *testing.T) {
+	setSidecarDBName("_vt")
+	var err error
+	origDefaultRdonly := defaultRdonly
+	defer func() {
+		defaultRdonly = origDefaultRdonly
+	}()
+	defaultRdonly = 0
+	vc = setupMinimalCluster(t)
+
+	err = vc.Vtctl.AddCellInfo("zone2")
+	require.NoError(t, err)
+	zone2, err := vc.AddCell(t, "zone2")
+	require.NoError(t, err)
+	require.NotNil(t, zone2)
+	defer vc.TearDown()
+
+	sourceKeyspace := "product"
+	targetKeyspace := "customer"
+	var mt iMoveTables
+	workflowName := "wf1"
+	targetTabs := setupMinimalCustomerKeyspace(t)
+
+	t.Run("MoveTablesCreateFlags1", func(t *testing.T) {
+		testMoveTablesFlags1(t, &mt, sourceKeyspace, targetKeyspace, workflowName, targetTabs)
+	})
+	t.Run("MoveTablesCreateFlags2", func(t *testing.T) {
+		testMoveTablesFlags2(t, &mt, sourceKeyspace, targetKeyspace, workflowName, targetTabs)
+	})
+	t.Run("MoveTablesCompleteFlags", func(t *testing.T) {
+		testMoveTablesFlags3(t, sourceKeyspace, targetKeyspace, targetTabs)
+	})
+}
+
+// Tests several create flags and some complete flags and validates that some of them are set correctly for the workflow.
+func testMoveTablesFlags1(t *testing.T, mt *iMoveTables, sourceKeyspace, targetKeyspace, workflowName string, targetTabs map[string]*cluster.VttabletProcess) {
+	tables := "customer,customer2"
+	createFlags := []string{"--auto-start=false", "--defer-secondary-keys=false", "--stop-after-copy",
+		"--no-routing-rules", "--on-ddl", "STOP", "--exclude-tables", "customer2",
+		"--tablet-types", "primary,rdonly", "--tablet-types-in-preference-order=true",
+		"--all-cells",
+	}
+	completeFlags := []string{"--keep-routing-rules", "--keep-data"}
+	switchFlags := []string{}
+	*mt = createMoveTables(t, sourceKeyspace, targetKeyspace, workflowName, tables, createFlags, completeFlags, switchFlags)
```

Comment:
There doesn't seem to be any reason to pass `mt` into this function. Is there? It's a pointer, but then we overwrite it with the address that `createMoveTables` returns as we dereference it. I'd remove that parameter and simplify this a bit.

Interfaces are pointers too so we don't need the double pointer (mt *iMoveTables) which means we have to use the somewhat awkward `(*mt)` notation. We can at least remove that part.